### ISSUE TITLE
[BUG] Variables in lists are not drawn correctly

### ISSUE THREAD

**Issue by metervara** [AdamRamberg/unity-atoms#104]
**Describe the bug**
When using variables in lists, _e.g. List\<FloatVariable\>_, the items are not drawn correctly once a variable has been assigned to an element in the list. Some items drawn by AtomDrawer & VariableDrawer are overlapping (see screenshot)

**To Reproduce**
1. Create a List of some Atom Variable in a MonoBehaviour
```
public class AtomDRawerIssue : MonoBehaviour
{
  // ...
  public List<FloatVariable> vars;
  // ...
}
```
2. Assign a FloatVariable to an element in the list in the inspector

**Expected behavior**
Assigned elements should be drawn without overlapping.

**Screenshots**
The first item in the list has been assigned and shows the overlapping issue:
<img width="506" alt="Screenshot 2020-02-10 at 22 02 39" src="https://user-images.githubusercontent.com/4705094/74190347-81fb9800-4c52-11ea-9470-7ecb8a771c07.png">

**Environment**
 - OS: OSX 10.14.5 (Mojave)
 - Unity Version 2019.3.0f3
 - Version of Unity Atoms v2.1.0

**Additional context**
I've tested with a few variable types but not all. Line 19 in VariableDrawer.cs seems to be part of the problem, the width is calculated using depth, and in the case of the list depth is not 0. Removing the depth condition on that line fixes the issue for Lists, but breaks drawing when using References.

**Comment by soraphis** [comment]
Thanks for reporting!
will have a look into it when I'm back home again. 

this is probably affects structs, too. 

I think we need to separate Reference-Drawers more from the Default drawers

**Comment by AdamRamberg** [comment]
Fixed in #109
